Cython/Utility/ImportExport.c

```diff
@@ -632,14 +597,14 @@ static int __Pyx_ImportVoidPtr_$cyversion(PyObject *module, const char *name, vo
 #endif
     if (!cobj) {
         PyErr_Format(PyExc_ImportError,
-            "%.200s does not export expected C variable %.200s",
-                PyModule_GetName(module), name);
+            "%.200s does not export expected C %s %.200s",
+                PyModule_GetName(module), what, name);
         goto bad;
     }
     if (!PyCapsule_IsValid(cobj, sig)) {
         PyErr_Format(PyExc_TypeError,
-            "C variable %.200s.%.200s has wrong signature (expected %.500s, got %.500s)",
-             PyModule_GetName(module), name, sig, PyCapsule_GetName(cobj));
+            "C %s %.200s.%.200s has wrong signature (expected %.500s, got %.500s)",
```

Comment:
String formatting is quicker if it knows the total buffer length upfront instead of calling `strlen()` on the arguments.
```suggestion
            "C %.8s %.200s.%.200s has wrong signature (expected %.500s, got %.500s)",
```